examples/ex1_2d_full_control.py

```diff
@@ -0,0 +1,183 @@
+from pathlib import Path
+import math
+import sklearn
+import inspect
+import sys
+import joblib
+import numpy as np
+from icarus import (DatasetGenerator,
+                    ModelBuilder,
+                    MooseSetup,
+                    UserInterface)
+
+def main():
+    """main: runs all of the other functions.
+    """
```

Comment:
By convention, docstrings should be in the imperative mood. This goes for all the docstrings in the codebase.
```suggestion
    """Run all of the other functions.
    """
```
See e.g. https://stackoverflow.com/questions/59902102/why-is-imperative-mood-important-for-docstrings

It's also better to give the user more information about what they're running. In isolation a function called main which "runs all the other functions" could be doing anything based on what you're telling the user here.